Initial library release

betzrhodes opened the pull request:
Agent library v1.0.0
Device library v0.1.0

Commit:
initial code

Commit:
added docs and comments

Commit:
configured helper functions for sffs file names

Commit:
added code commentss

Commit:
fixed typo, updated write ubx assist method name

Commit:
added links to readme

Commit:
fixed invalid protocol error bug

Commit:
Some review comments, bug fixing and clean up.

Commit:
Minor.

Commit:
cleanup after reveiw

Commit:
added usage limit note

Commit:
Update AGENT_LIB_README.md

Commit:
minor clean up

Commit:
fixed merge conflict

Commit:
examples added and tested

Commit:
added example readme

Commit:
updated docs, change dev ver, updated example agent

Commit:
example and doc cleanup

Commit:
added tests

Commit:
updated device library removed sffs

Commit:
added timeout to assist write, added tests

Commit:
added tests, minor error handling code updates

Commit:
fixed example code

Commit:
added set write timeout

Commit:
added timeout test

Commit:
added JSON encoder

Commit:
fixed typo in logger

smittytone committed:
Initial edits

smittytone committed:
Fix links

smittytone commented:
FYI, I've added the method _getMgaAck()_ to the Device Read Me as it's referenced in the copy. But it's not implemented in the source code. Let me know if you're not going to implement this, and I'll remove the method description.

betzrhodes commented:
> FYI, I've added the method _getMgaAck()_ to the Device Read Me as it's referenced in the copy. But it's not implemented in the source code. Let me know if you're not going to implement this, and I'll remove the method description.

I've removed the _getMgaAck() - 
When the assist messages are written there are hundreds of acks received. There is no way to bubble these up to the user in any meaningful way. The messages are parsed by the library and if they contain errors the parsed errors are bubbled up to the user via the onDone callback. It would only be confusing to add a getMgaAck to the the library.

Commit:
doc updates

smittytone committed:
Style fixes

Commit:
updated examples readme

Commit:
Merge branch 'develop' of github.com:electricimp/UBloxAssistNow into develop

Commit:
updated writeUtcTimeAssist to validate year without parameter

Commit:
fixed typo
